documentation/manual/src/doc/index.md

```diff
@@ -1,33 +1,55 @@
 endpoints
 =========
 
-*endpoints* is a Scala library for defining communication protocols over HTTP between
-applications.
-
-Noteworthy features:
-
-- endpoints descriptions are **first-class Scala values**, which can be reused,
-  combined and abstracted over ;
-- a *same* endpoint description can be interpreted in *multiple* ways:
-    - a **client** implementation (JVM and/or Scala.js),
-    - a **server** implementation,
-    - an **OpenAPI** definition file,
-    - thus, your client and documentation are always **consistent** with the actual
-      implementation of your server,
-- **type safety**: you get a compile-time error if you invoke an endpoint but supply incorrect data
-  (requests are guaranteed to be well constructed),
-- **extensibility**: you can introduce both
-  - new *descriptions* that are specific to your application (e.g. the usage
-    of a particular HTTP header),
-  - new *interpreters* for endpoint descriptions (e.g. generation of a RAML documentation),
-- **vanilla Scala**: being written in pure Scala (no macros), the implementation is easy to
-  reason about and IDE friendly.
+*endpoints* is a Scala library for remote communication. It improves the developers productivity by ensuring
+that HTTP servers, HTTP clients, and documentation always agree on the same protocol.
+
+- **maintenance is simplified**: the API documentation is automatically updated when an endpoint is modified ;
+- errors are raised at **compile-time** if endpoints are invoked with incompatible parameters.
+
+Servers, clients and documentation are *all* derived from a *single* source of truth describing the underlying
+protocol details (e.g., which verb, path, query parameters, headers, authentication strategy, etc. to
+use). For instance, here is an endpoint for incrementing a counter. It uses the HTTP verb `POST`, the URL path
+`/increment`, a JSON request entity containing an `Increment` value, and it returns an empty response.
+
+~~~ scala src=../../../../documentation/examples/quickstart/endpoints/src/main/scala/quickstart/CounterEndpoints.scala#endpoint-definition
+~~~
+
+From the client perspective, calling an HTTP endpoint is as simple as calling a function:
+
+~~~ scala src=../../../../documentation/examples/quickstart/client/src/main/scala/quickstart/CounterClientFuture.scala#endpoint-invocation
+~~~
+
+Conversely, from the server perspective implementing an HTTP endpoint is as simple as implementing a function:
+
+~~~ scala src=../../../../documentation/examples/quickstart/server/src/main/scala/quickstart/CounterServer.scala#endpoint-implementation
+~~~
```

Comment:
Here is what it renders to:

![image](https://user-images.githubusercontent.com/332812/61656769-c7aa4080-acc1-11e9-957c-627b39edf421.png)

Reply:
nice :+1: